src/data/types.ts

```diff
@@ -90,6 +92,130 @@ export interface StreamInfo {
   viewer_count: number;
 }
 
+export interface StreamAnnouncementInfo {
+  id: string;
+  category_name: string;
+  language: string;
+  started_at: Date;
+  thumbnail_url: string;
+  title: string;
+  streamer_info: StreamerInfo;
+  viewer_count: number;
+  streaming_service: StreamingService;
+  online: boolean;
+  archivedStream?: VideoByUserIdResponse;
+}
+
+
+export interface StreamingService {
+  name: string;
+  base_url: string;
+}
+
+export const Glimesh: StreamingService = {
+  name: "Glimesh",
+  base_url: "https://glimesh.tv/"
+} as const
+
+export const Twitch: StreamingService = {
+  name: "Twitch",
+  base_url: "https://twitch.tv/"
+} as const
+
+export interface StreamAnnouncementInfoResolver {
+  resolve(): Promise<StreamAnnouncementInfo>;
+}
+export class TwitchStreamInfoResolver
+  implements StreamAnnouncementInfoResolver
+{
+  constructor(private streamInfo: StreamInfo) {}
+
+  async resolve(): Promise<StreamAnnouncementInfo> {
+    const streamer = await UserManager.getUserAsStreamerInfoById(this.streamInfo.user_id);
+    let category = await fetchGameById(this.streamInfo.game_id);
+    if (!category) {
+      category = { name: "" };
+    }
+    const started_at = new Date(this.streamInfo.started_at);
+
+    return {
+      id: this.streamInfo.id,
+      title: this.streamInfo.title,
+      category_name: category.name,
+      language: this.streamInfo.language,
+      started_at: started_at,
+      thumbnail_url: this.streamInfo.thumbnail_url,
+      viewer_count: this.streamInfo.viewer_count,
+      streamer_info: streamer,
+      streaming_service: Twitch,
+      online:true
+    };
+  }
+}
+
+export class GlimeshStreamInfoProvider
+  implements StreamAnnouncementInfoResolver
+{
+  constructor(private streamInfo: GlimeshStreamInfo) {} // private streamInfo: GlimeshStreamInfo) {}
+  async resolve(): Promise<StreamAnnouncementInfo> {
+    return {
+      category_name: this.streamInfo.stream?.category.name,
```

Comment:
Can we match the order of keys in this object and the one from above? Reduces the cognitive overhead of trying to figure out if they are the same shape. Alphabetical is best, to make it easier to update these in the future.

Not sure if I like the fact these are duplicated, or if we can somehow unify these a bit more, mainly for ease of maintenance in the future.